test/signature_parser.cpp

```diff
@@ -12,7 +12,31 @@ BOOST_AUTO_TEST_CASE(input_signature_parser_simple) {
     BOOST_TEST(signature_parser().parse("struct"));
     BOOST_TEST(signature_parser().parse("struct<field>"));
     BOOST_TEST(signature_parser().parse("array<int>"));
+    BOOST_TEST(signature_parser().parse("field"));
+    BOOST_TEST(signature_parser().parse("field<pallas_base>"));
+    BOOST_TEST(signature_parser().parse("curve<pallas>"));
+    BOOST_TEST(signature_parser().parse("int"));
+    BOOST_TEST(signature_parser().parse("array<int>"));
+    BOOST_TEST(signature_parser().parse("array<field>"));
+    BOOST_TEST(signature_parser().parse("array<field<pallas_base>>"));
+    BOOST_TEST(signature_parser().parse("struct<int>"));
```

Comment:
It looks like all the tests that contain `struct<` are failing now with error `Error when parsing signature: expecting "," before: ">"`